### DISCUSSION TITLE
Vote: BlockOperator interface

### DISCUSSION THREAD

**Discussion by tzanio** [mfem/mfem#4223] [Polls]
There has been a series of issues around the const-ness of the elements of `BlockOperator`: #4078, #4135 and #4136. Lets vote below on how to proceed.

**Comment by najlkin** [comment]
Maybe a certain compromise solution, which I do not like much 😕 , but just offering 😉. We could introduce `TBlockOperator` internally as an implementation of `Block(Const)Operator` (avoiding duplicity of the code) without exposing the template to users if we do not trust them to use it correctly 🤔 .

**Comment by ebchin** [comment]
Maybe there should be an option "Revert the const changes introduced in #4078 and merge #4135"?

**Reply by tzanio** [reply]
Sorry, the second option should not have been in the poll -- https://github.com/mfem/mfem/pull/4135 is orthogonal to the two choices and we probably want to merge it either way.
